packages/docs/guide/advanced/transitions.md

```diff
@@ -72,13 +73,18 @@ router.afterEach((to, from) => {
 
 Vue might automatically reuse components that look alike, avoiding any transition. Fortunately, it is possible [to add a `key` attribute](https://vuejs.org/api/built-in-special-attributes.html#key) to force transitions. This also allows you to trigger transitions while staying on the same route with different params:
 
+```js
+import { useRoute } from 'vue-router'
+const route = useRoute()
+```
+The `router-view` needs the route's full path as key. Although the route can be destructured from the slot props, that destructured route cannot be used in the key attribute. That's why getting the route from `useRoute` is the clean code solution. Setting a `key` on the inner `component` tag may cause application crashes in certain situations. See discussion in [issue 2121](https://github.com/vuejs/router/issues/2121) for more information about that.
+
 ```vue-html
-<router-view v-slot="{ Component, route }">
+<router-view v-slot="{ Component }" :key="route.fullPath">
```

Comment:
Thank you, but this is, most of the time, harmful in different ways:

- fullPath changes more than just path
- this doesn't work with transitions

On top of that, using `useRoute()` here breaks in other scenarios where we can force the resolved route for a router-view

Reply:
What about just using `route.path`? Would it solve your concerns?

The point of my changes was to make it work with transitions. Could you please point out how it is not working? The linked Stackblitz project at the end of the PR is using transitions and they are all working.

Reply:
No, because the `key` needs to be on the component. Otherwise, the whole thing is being replaced. In your case, it works because of `appear`, but that shouldn't be necessary in the first place

Reply:
I will close this for now but feel free to open a new PR with a fix for #2121. Note there shouldn't be a need to change the current way of doing things.